virtcontainers/clh.go

```diff
@@ -421,28 +378,58 @@ func (clh *cloudHypervisor) addDevice(devInfo interface{}, devType deviceType) e
 	defer span.Finish()
 
 	var err error
+	devLogger := clh.Logger().WithFields(log.Fields{
+		"function": "addDevice",
+	})
 
 	switch v := devInfo.(type) {
 	case Endpoint:
-		clh.Logger().WithField("function", "addDevice").Infof("Adding Endpoint of type %v", v)
-		clh.cliBuilder.AddNet(CLINet{
-			device: v.Name(),
-			mac:    v.HardwareAddr(),
-		})
+		devLogger.Infof("Adding Endpoint of type %v", v)
+		n := chclient.NetConfig{}
+		n.Mac = v.HardwareAddr()
+		n.Tap = v.NetworkPair().TapInterface.TAPIface.Name
+
+		//TODO server does not allow empty IP fail serde to get it
+		n.Ip = "0.0.0.0"
+		n.Mask = "0.0.0.0"
+		clh.vmconfig.Net = append(clh.vmconfig.Net, n)
+
 	case types.HybridVSock:
-		clh.Logger().WithFields(log.Fields{
-			"function": "addDevice",
-			"path":     v.UdsPath,
-			"cid":      v.ContextID,
-			"port":     v.Port,
+		devLogger.WithFields(log.Fields{
+			"path": v.UdsPath,
+			"cid":  v.ContextID,
+			"port": v.Port,
 		}).Info("Adding HybridVSock")
-		clh.cliBuilder.SetVsock(&CLIVsock{
-			cid:        uint32(v.ContextID),
-			socketPath: v.UdsPath,
-			iommu:      false,
-		})
+		clh.vmconfig.Vsock = []chclient.VsockConfig{{Cid: defaultGuestVSockCID, Sock: v.UdsPath}}
+	case types.Volume:
+
+		if clh.config.SharedFS != config.VirtioFS {
+			return fmt.Errorf("shared fs method not supported %s", clh.config.SharedFS)
+		}
+
+		// set the virtio-fs to the hypervisor
+		vfsdSockPath, err := clh.virtioFsSocketPath(clh.id)
+		if err != nil {
+			return err
+		}
+
+		clh.vmconfig.Fs = []chclient.FsConfig{
+			{
+				Tag:       v.MountTag,
+				CacheSize: int64(clh.config.VirtioFSCacheSize << 20),
+				Sock:      vfsdSockPath,
+				//TODO
+				NumQueues: 1,
+				//TODO
+				QueueSize: 512,
+				//TODO dax
+			},
+		}
+		devLogger.Debug("Adding Volume to hypervisor ", v.HostPath, ":", v.MountTag)
```

Comment:
1. A few TODO's here?
2. perhaps would be cleaner to add these to their own functions? Ie "clhAddVolume" ?